packages/frametricks/init.lua

```diff
@@ -153,12 +153,28 @@ function package:registerCommands ()
 
   self:registerCommand("showframe", function (options, _)
     local id = options.id or SILE.typesetter.frame.id
-    if id == "all" then
-      for _, frame in pairs(SILE.frames) do
-        SILE.outputter:debugFrame(frame)
+    options.allpages = options.allpages or true
+
+    local show = function()
+      if id == "all" then
+        for _, frame in pairs(SILE.frames) do
+          SILE.outputter:debugFrame(frame)
+        end
+      else
+        SILE.outputter:debugFrame(SILE.getFrame(id))
+      end
+    end
+
+    if options.allpages and options.allpages ~= "false" then
+      show()
+      local oldNewPage = SILE.documentState.documentClass.newPage
```

Comment:
No, don't do this. I know we do funky things like this elsewhere but we're trying to move *away* from this, not towards it. Set a new page hook instead and just leave the `class:newPage()` function alone.

Reply:
Sorry, I don't know how to play with hooks for now. I read this part of the documentation and searched for uses of it in this repository, but still can't manage to set hooks around there... Can you give me at least an example?

Reply:
Sure, happy to help. I think in this case this whole section of code can be reduced as in [this code suggestion comment](https://github.com/sile-typesetter/sile/pull/1909/files#r1390349519).